torch/optim/lr_scheduler.py

```diff
@@ -195,6 +195,15 @@ def step(self, epoch=None):
         self._last_lr = [group["lr"] for group in self.optimizer.param_groups]
 
 
+def _warn_get_lr_called_within_step(lr_scheduler: LRScheduler):
+    if not lr_scheduler._get_lr_called_within_step:
+        warnings.warn(
+            "To get the last learning rate computed by the scheduler, "
+            "please use `get_last_lr()`.",
+            UserWarning,
```

Comment:
```suggestion
            UserWarning,
            stacklevel=2
```
We should report the warning on the get_lr call that the user is making, not this internal one 😛

Reply:
fixed